lambdatimes/src/components/Carousel/Carousel.js

```diff
@@ -1,33 +1,70 @@
 import React, { Component } from 'react';
 import { carouselData } from '../../data'
-// Complete this Carousel 
+// Complete this Carousel
 export default class Carousel extends Component {
-  constructor(props){
+  constructor(props) {
     super(props);
     this.state = {
-
+      carouselData: carouselData,
+      index: 0
     }
   }
-  componentDidMount(){
 
+  componentDidMount() {
+    this.previewCarousel();
   }
 
   leftClick = () => {
+    let index = this.state.index;
+    const imagesLength = this.state.carouselData.length;
 
+    if (index === 0) {
+      index = imagesLength - 1;
+      this.setState({ index });
+    } else {
+      index -= 1;
+      this.setState({ index });
+    }
   }
 
   rightClick = () => {
+    let index = this.state.index;
+    const imagesLength = this.state.carouselData.length;
+
+    if (index === imagesLength - 1) {
+      index = 0;
+      this.setState({ index });
+    } else {
+      index += 1;
+      this.setState({ index });
+    }
+  }
 
+  selectedImage = (idx) => {
+    return <img src={this.state.carouselData[idx]} style={{ display: 'block' }} alt="img" />
   }
 
-  selectedImage = () => {
-    return <img src={} style={{display: 'block'}} />
+  previewCarousel = () => {
+    let index = this.state.index;
+    const imagesLength = this.state.carouselData.length;
```

Comment:
Amazing work on the carousel stretch goal! I like how you even got the infinite scroll working! 💪